test/e2e/idle_connection_test.go

```diff
@@ -0,0 +1,778 @@
+		Spec: corev1.ServiceSpec{
+			Selector: selectorLabels,
+			Ports: []corev1.ServicePort{{
+				Name:       "http",
+				Port:       8080,
+				TargetPort: intstr.FromInt32(8080),
+				Protocol:   corev1.ProtocolTCP,
+			}},
+		},
+	}
+
+	if err := kclient.Create(ctx, service); err != nil {
+		return nil, fmt.Errorf("failed to create service %s/%s: %w", service.Namespace, service.Name, err)
+	}
+
+	return service, nil
+}
+
+func idleConnectionCreateRoute(ctx context.Context, namespace, name, serviceName string, labels map[string]string) (*routev1.Route, error) {
+	route := &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    labels,
+		},
+		Spec: routev1.RouteSpec{
+			Subdomain: name,
+			To: routev1.RouteTargetReference{
+				Kind: "Service",
+				Name: serviceName,
+			},
+			Port: &routev1.RoutePort{
+				TargetPort: intstr.FromString("http"),
+			},
+			WildcardPolicy: routev1.WildcardPolicyNone,
+		},
+	}
+
+	if err := kclient.Create(ctx, route); err != nil {
+		return nil, fmt.Errorf("failed to create route %s/%s: %w", route.Namespace, route.Name, err)
+	}
+
+	return route, nil
+}
+
+func idleConnectionSwitchRouteService(t *testing.T, ic *operatorv1.IngressController, tc *idleConnectionTestConfig, routeName types.NamespacedName, serviceIndex int) (*routev1.Route, error) {
+	if serviceIndex >= len(tc.services) {
+		return nil, fmt.Errorf("service index %d out of range", serviceIndex)
+	}
+
+	service := tc.services[serviceIndex]
+
+	var updatedRoute *routev1.Route
+	if err := updateRouteWithRetryOnConflict(t, routeName, time.Minute, func(route *routev1.Route) {
+		route.Spec.To.Name = service.Name
+		updatedRoute = route
+	}); err != nil {
+		return nil, fmt.Errorf("failed to update route %s to point to service %s/%s: %w", routeName, service.Namespace, service.Name, err)
+	}
+
+	t.Logf("Switched route %s to service %s/%s", routeName, service.Namespace, service.Name)
+
+	routeAdmittedCondition := routev1.RouteIngressCondition{
+		Type:   routev1.RouteAdmitted,
+		Status: corev1.ConditionTrue,
+	}
+	if err := waitForRouteIngressConditions(t, kclient, routeName, ic.Name, routeAdmittedCondition); err != nil {
+		return nil, fmt.Errorf("error waiting for route to be admitted: %w", err)
+	}
+
+	t.Logf("Route %s admitted by ingresscontroller %s", routeName, ic.Name)
+
+	expectedBackendName := fmt.Sprintf("be_http:%s:%s", routeName.Namespace, routeName.Name)
+	expectedServerName := fmt.Sprintf("pod:%s:%s:http:%s:%d", tc.pods[serviceIndex].Name, service.Name, tc.pods[serviceIndex].Status.PodIP, service.Spec.Ports[0].Port)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+	if err := waitForHAProxyConfigUpdate(ctx, t, ic, expectedBackendName, expectedServerName); err != nil {
+		return nil, fmt.Errorf("error waiting for HAProxy configuration update for route %s to point to service %s/%s: %w", routeName, service.Namespace, service.Name, err)
+	}
```

Comment:
Yes, all good points. Given this is a parallel test I could just sleep 30s (for busy clusters).

Reply:
I will rip this out and sleep 30s instead. Why 30s? To avoid flakes if we chose 6s (i.e., a 1s more than the 5s routing update default). There's little point in chasing test flakes when the test is a) parallel and b) runs for a couple of minutes anyway.